### PR TITLE
Work around problems introduced by GH-1305

### PR DESCRIPTION
**Pull request by cpitclaudel**
Haven't had much time to test this out yet.
Let's look at it after we fix the test suite (thanks for the hard work @fmdkdd!)

### CODE REVIEW

**Comment by fmdkdd** [comment]
You'll need to rebase this on master to fix the CI.

File: flycheck.el [hunk 1]
```diff
@@ -2745,14 +2745,21 @@ buffer."
           (flycheck-buffer-automatically 'new-line 'force-deferred)
         (setq flycheck-idle-change-timer
               (run-at-time flycheck-idle-change-delay nil
-                           #'flycheck-handle-idle-change (current-buffer)))))))
+                           #'flycheck--handle-idle-change (current-buffer)))))))
 
-(defun flycheck-handle-idle-change (buffer)
-  "Handle an expired idle timer in BUFFER since the last change."
+(defun flycheck--handle-idle-change (buffer)
```
Comment (fmdkdd) [inline flycheck.el]: LGTM, except that I would like the wrapper to have a more discerning name than just an added dash.  Maybe add `-internal` or `-in-buffer` to the end?

Comment (cpitclaudel) [reply flycheck.el]: Good point :) Updated.

**Comment by fmdkdd** [comment]
@cpitclaudel LGTM.  Though I'm wondering if we shouldn't add a test to catch regressions on this one.  It looks like we already have a couple of tests related to idle-change in `flycheck-test.el` that could serve as a base.

**Comment by cpitclaudel** [comment]
Possibly. I'm between trips ATM, so I have very little time. I'd vote for merging this now since the bug is already impacting people, and opening an issue for the tests.  I can tackle the issue in 2 weeks.

**Comment by fmdkdd** [comment]
@cpitclaudel Good for me.  See #1324.

**Pull request merged**